lightkurve/periodogram.py

```diff
@@ -1,99 +1,947 @@
+allows us to keep track of what units our data are in, and easily switch
+between different units."""
+from astropy import units as u
+
+from . import PACKAGEDIR, MPLSTYLE
+
+log = logging.getLogger(__name__)
 
-__all__ = ['Periodogram']
+__all__ = ['Periodogram', 'estimate_mass', 'estimate_radius',
+            'estimate_mean_density', 'stellar_params', 'standardize_units']
 
+###Add uncertainties <- review these [astero]
+numax_s = 3090.0 # Huber et al 2013
+err_numax_s = 30.
+deltanu_s = 135.1
+err_deltanu_s = 0.1
+teff_s = 5777.0
+
+#To_do: Parition this into a Periodogram object and a LombScarglePeriodogram
+#object that inherits from periodogram
 
 class Periodogram(object):
+    ###Update this [houseekeping]
     """Represents a power spectrum.
-
     Attributes
     ----------
-    frequencies : array-like
-        List of frequencies.
-    powers : array-like
-        Power measurements.
+    lc : `LightCurve` object
+        The LightCurve from which the Periodogram is computed. Holds all  of
+        the `LightCurve object's properties.
+    frequency : array-like
+        List of frequencies with associated astropy unit.
+    period : array-like
+        List of periods (1 / frequency) with associated astropy unit
+    power : array-like
+        The power-spectral-density of the Fourier timeseries, in units of
+        ppm^2 / freq_unit, where freq_unit is the unit of the frequency
+        attribute.
+    nyquist : float
+        The Nyquist frequency of the lightcurve. In units of freq_unit, where
+        freq_unit is the unit of the frequency attribute.
+    frequency_spacing : float
+        The frequency spacing of the periodogram. In units of freq_unit, where
+        freq_unit is the unit of the frequency attribute.
+    max_power : float
+        The power of the highest peak in the periodogram
+    max_pow_frequency : float
+        The frequency corresponding to the highest power in the periodogram
+    max_pow_period : float
+        The periodod corresponding to the highest power in the periodogram.
+    _format : str
+        {'frequency', 'period'}. Default 'frequency'. Determines the format
+        of the periodogram. If 'frequency', preferred units will be frequency.
+        If 'period', preferred units will be period.
+    meta : dict
+        Free-form metadata associated with the Periodogram.
     """
-    def __init__(self, frequencies=None, powers=None):
-        self.frequencies = frequencies
-        self.powers = powers
+    def __init__(self, lc=None, frequency=None, period = None, power=None,
+                nyquist=None, fs=None, _format = 'frequency', meta={}):
+        self.lc = lc
+        self.frequency = frequency
+        self.period = period
+        self.power = power
+        self.nyquist = nyquist
+        self.frequency_spacing = fs
+        self._format = _format
+        self.meta = meta
+
+        if self.period is None:
+            self.period = 1./self.frequency
```

Comment:
Will we also convert period to frequency if `self.frequency is None`, or should we make `self.frequency` a required argument?  (In which case it should not default to None.)

Reply:
Frequency is now a required argument (as is Power). Period is a calculated property as 1/frequency.